### PR TITLE
move helper funcs to util/deployment.go from util.go

### PR DESCRIPTION
**Pull request by mqliang**
clean up:

helper funcs `GetIntOrPercentValue()` and `GetValueFromPercent()` should be in util/deployment.go instead of  util.go

@bgrant0607 @janetkuo

### CODE REVIEW

**Comment by mqliang** [comment]
After carefully looking, I think we should:
1) move helper funcs `GetIntOrPercentValue()` and `GetValueFromPercent()` to pkg/util/intstr/intstr.go
2) move pkg/util/deployment.go  to /pkg/controller/deployment/deployment_util.go

Any thought?

**Comment by bgrant0607** [comment]
As long as you're doing this, could you create `GetValueFromIntOrPercent(string, int)`? I've commented on this a few places, such as:
https://github.com/kubernetes/kubernetes/pull/20696/files?diff=unified#r52562178
https://github.com/kubernetes/kubernetes/pull/20273/files#r51953975

**Comment by bgrant0607** [comment]
@mqliang I haven't checked, but I thought the motivation to put code in /util rather than just /controller was that it was shared between the controller and the client. If it really is just used by the controller, we could move it there.

**Comment by bgrant0607** [comment]
cc @kargakis

**Comment by mqliang** [comment]
@bgrant0607 Updated. PTAL

**Comment by kargakis** [comment]
Seems the deployment controller is the only user of those helpers

LGTM

File: pkg/util/intstr/intstr.go [hunk 1]
```diff
@@ -113,3 +115,29 @@ func (intstr *IntOrString) Fuzz(c fuzz.Continue) {
 		c.Fuzz(&intstr.StrVal)
 	}
 }
+
+func GetValueFromIntOrPercent(intOrStr *IntOrString, value int) (int, error) {
+	maxSurge, isPercent, err := getIntOrPercentValue(intOrStr)
```
Comment (bgrant0607) [inline pkg/util/intstr/intstr.go]: This isn't necessarily maxSurge. The same pattern applies to maxAvailable.

Besides, this is a generic utility.

I would rename `intOrString` to `intOrPercent`, `value` to `total`, and `maxSurge` to `value`.

**Comment by bgrant0607** [comment]
I guess these functions didn't have any unit tests? Could you please create a simple one?

**Comment by bgrant0607** [comment]
LGTM, thanks much

**Comment by mqliang** [comment]
@bgrant0607 Rebased. Could you please reapply the LGTM tag?

**Comment by bgrant0607** [comment]
LGTM. Sorry about the rebases.

**Pull request merged**